main.go

```diff
@@ -499,26 +531,46 @@ func runKubectlLabelOrTaint(ctx *Context, node string, kctlCmd string, labelsOrT
 	return nil
 }
 
-func installCertManager(ctx *Context) error {
-	fmt.Println("Installing cert manager...")
-	cmd := exec.Command("kubectl", "apply", "-f", "https://github.com/jetstack/cert-manager/releases/download/v1.11.1/cert-manager.yaml")
+func runKubectlApply(ctx *Context, applyFlag string, url string) error {
+	cmd := exec.Command("kubectl", "apply", applyFlag, url)
 	if _, err := runCommand(ctx, cmd); err != nil {
 		return err
 	}
-
 	return nil
 }
 
-func installMPIOperator(ctx *Context) error {
-	fmt.Println("Installing mpi-operator...")
-	cmd := exec.Command("kubectl", "apply", "-f", "https://raw.githubusercontent.com/kubeflow/mpi-operator/v0.4.0/deploy/v2beta1/mpi-operator.yaml")
-	if _, err := runCommand(ctx, cmd); err != nil {
+func runKubectlApplyK(ctx *Context, url string) error {
+	return runKubectlApply(ctx, "-k", url)
+}
+
+func runKubectlApplyF(ctx *Context, url string) error {
+	return runKubectlApply(ctx, "-f", url)
+}
+
+func installCertManager(ctx *Context) error {
+	fmt.Println("Installing cert manager...")
+	err := runKubectlApplyF(ctx, "https://github.com/jetstack/cert-manager/releases/download/v1.11.1/cert-manager.yaml")
+	if err != nil {
+		return err
+	}
+	// The NNF services will dump errors at installation time if the
+	// cert-manager webhook isn't ready.
+	fmt.Println("  waiting for its webhook to be ready...")
+	waiter := "kubectl wait pod -n cert-manager --timeout=180s -l app.kubernetes.io/component=webhook --for jsonpath='{.status.phase}'=Running"
+	cmd := exec.Command("bash", "-c", waiter)
+	_, err = runCommand(ctx, cmd)
+	if err != nil {
+		fmt.Printf("\n\033[1mThe cluster is still waiting for cert-manager to start.\nPlease run `nnf-deploy init` after cert-manager is running.\033[0m\n\n")
 		return err
 	}
-
 	return nil
 }
 
+func installMPIOperator(ctx *Context) error {
+	fmt.Println("Installing mpi-operator...")
+	return runKubectlApplyF(ctx, "https://raw.githubusercontent.com/kubeflow/mpi-operator/v0.4.0/deploy/v2beta1/mpi-operator.yaml")
```

Comment:
Would we want to store the path to the yamls for `mpi-operator` and `cert-manager` in repositories.yaml?

Reply:
That's tempting.  Maybe in the next PR.